cli/azd/cmd/util.go

```diff
@@ -169,94 +186,77 @@ func ensureEnvironmentInitialized(ctx context.Context, environmentName string, e
 	}
 
 	if !hasEnvName {
-		env.SetEnvName(environmentName)
+		env.SetEnvName(envSpec.environmentName)
 	}
 
-	if !hasSubID || !hasPrincipalID || !hasLocation {
+	needAzureInteraction := (!hasSubID && envSpec.subscription == "") ||
```

Comment:
I wonder if we can simplify these conditionals and some of the double checking logic later by doing something like this:

```go
applyDefault := func(key string, value string) { 
  if _, has := env.Values[key]; !has {
      env.Values[key] = value
  }
}
```

and then call `applyDefault` to copy the values from `envSpec` into the environment before we do the `hasValue` checks.

Reply:
Agreed, it got a bit convoluted. LMK how you like my latest attempt!

Reply:
(resolving since you approved the PR)